### PR TITLE
Allow plugins to specify a ReleaseNotesURL via manifest

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This allows pre packaged and local plugins to set `ReleaseNotesURL`. I also did update the test bundles.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21868

### CODE REVIEW

File: model/manifest.go [hunk 1]
```diff
@@ -330,6 +333,10 @@ func (m *Manifest) IsValid() error {
 		return errors.New("invalid SupportURL")
 	}
 
+	if m.SupportURL != "" && !IsValidHttpUrl(m.ReleaseNotesURL) {
```
Comment (hanzei) [inline model/manifest.go]: The logic of the other checks is wrong. The community member who submitted the code, will fix it. See https://github.com/mattermost/mattermost-server/issues/13761

Comment (marianunez) [reply model/manifest.go]: Shouldn't the check here for empty be for `ReleaseNotesURL`?
```suggestion
	if m.ReleaseNotesURL != "" && !IsValidHttpUrl(m.ReleaseNotesURL) {
```

**Comment by prapti** [comment]
@hanzei In the initial observation, as soon as I switch my branch to this PR and upload Matterpoll 1.3.0, the following error shows up in the logs:
```
DEBU[2020-01-27T19:00:16.072963-08:00] using plugin                                  caller="plugin/hclog_adapter.go:33" plugin_id=com.github.matterpoll.matterpoll wrapped_extras=version1
WARN[2020-01-27T19:00:16.0815291-08:00] This looks to be a fresh install. Setting database schema version to 1.3.0.  caller="mlog/sugar.go:27" plugin_id=com.github.matterpoll.matterpoll
ERRO[2020-01-27T19:00:16.104264-08:00] SqlUserStore.Get: We encountered an error finding the account, SqlUserStore.Get: Unable to find the user., user_id=com.github.matterpoll.matterpoll  caller="app/bot.go:35"
DEBU[2020-01-27T19:00:17.504195-08:00] plugin process exited                         caller="plugin/hclog_adapter.go:33" plugin_id=com.github.matterpoll.matterpoll wrapped_extras=pathplugins/com.github.matterpoll.matterpoll/server/dist/plugin-darwin-amd64pid24609
DEBU[2020-01-27T19:00:17.504243-08:00] plugin exited                                 caller="plugin/hclog_adapter.go:35" plugin_id=com.github.matterpoll.matterpoll
ERRO[2020-01-27T19:00:17.509608-08:00] Unable to restart plugin on upgrade.          caller="mlog/log.go:175" err_details="failed to ensure bot user: failed to create bot: SqlUserStore.Get: We encountered an error finding the account, SqlUserStore.Get: Unable to find the user., user_id=com.github.matterpoll.matterpoll" err_where=installExtractedPlugin http_code=500 ip_addr="::1" method=POST path=/api/v4/plugins request_id=rwifrt8sqtr57x8qqrbeso6yuy user_id=554whw843t8p5jdsuqgro5rkba
```
However, I can verify the QA test steps in the [ticket](https://mattermost.atlassian.net/browse/MM-21868) upon refreshing.

**Comment by hanzei** [comment]
@prapti Good observation, thanks for sharing it. This is unrelated to this PR and also happens on `master`. I filled https://mattermost.atlassian.net/browse/MM-22106 to track it.

**Comment by aaronrothschild** [comment]
This process is documented internally for when a new plugin is added to the marketplace on this checklist here: https://github.com/mattermost/mattermost-marketplace/blob/production/.github/ISSUE_TEMPLATE/add_plugin.md

**Pull request merged**